doc/api_reference.md

```diff
@@ -349,6 +350,28 @@ cancellation of the operation returned by the function. Note that cancellation
 may also be requested through the stop-token of the receiver that is connected
 to the sender returned by `let_value_with_stop_source()`.
 
+### `let_value_with_stop_token(Invocable func) -> Sender`
+
+The `let_value_with_stop_token()` algorithm creates an
+`inplace_stop_token_adapter` that remains alive for the duration of an
+operation.
+
+`func` is invoked with an lvalue reference to an `inplace_stop_token`
+created by subscribing the `inplace_stop_token_adapter` to the stop token
+returned by the invocation of `get_stop_token` on the provided receiver.
+This invocation must return a Sender.
+
+The reference passed to `func` will remain valid until the returned sender
+completes, at which point the `inplace_stop_token_adapter` goes out of scope.
```

Comment:
I don't think the documentation here needs to refer to the use of `inplace_stop_token_adapter` as that is an implementation detail.

It just needs to say something along the lines of:
> The `let_value_with_stop_token()` algorithm takes a function object that is invoked at `unifex::connect()` time with an `inplace_stop_token` object that can be used to receive a stop-request sent by the parent operation through the receiver it passes to `connect()`.
>
> The function must return a sender, which is immediately connected. The result of the sender returned from the function becomes the result of the `let_value_with_stop_token()` sender.
>
> The function is guaranteed to be kept alive until the operation completes.
>
> The stop-token passed to the function is only guaranteed to be valid until the sender returned by the function completes.